File: app/src/main/java/com/zcy/designsupportdemo/adapter/VPAdapter.java
package com.zcy.designsupportdemo.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

/**
 * Created by zcy on 2017/9/21.
 */

public class VPAdapter extends FragmentPagerAdapter {

    private List<Fragment> mFragments;
    private String[] mTitles;

    public VPAdapter(FragmentManager fm, List<Fragment> fragments,String[] titles) {
        super(fm);
        mFragments = fragments;
        mTitles=titles;
    }

    @Override
    public Fragment getItem(int position) {
        return mFragments.get(position);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTitles[position];
    }

    @Override
    public int getCount() {
        return mFragments.size();
    }


}

File: app/src/main/java/com/zcy/designsupportdemo/adapter/VPAdapter.kt
package com.zcy.designsupportdemo.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * Created by zcy on 2017/9/21.
 */
//在主构造函数里，可以将参数声明成类的属性。
//也可以用这些参数变量赋值给类的属性。
class VPAdapter(fm: FragmentManager, private val mFragments: List<Fragment>, private val mTitles: Array<String>) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return mFragments[position]
    }

    override fun getPageTitle(position: Int): CharSequence {
        return mTitles[position]
    }

    override fun getCount(): Int {
        return mFragments.size
    }


}
